### DISCUSSION TITLE
Reduce astro package install size?

### DISCUSSION THREAD

**Discussion by Siilwyn** [withastro/roadmap#100] [Proposal]
The package install size [190 MB](https://packagephobia.com/result?p=astro). Reducing the size will help install speed and save some bandwidth. To put things in perspective, [Nuxt & Next are both pretty big](https://packagephobia.com/result?p=astro%2Cnext%2Cnuxt) too, around 100 MB.

The biggest directories when installed locally (it's a bit bigger than packagephobia):

![image](https://user-images.githubusercontent.com/5701149/152865477-50553824-cc59-44c7-88d6-8c595481a6db.png)

**I'm mostly surprised by the fact that astro includes all possible renderers!** That would be a straightforward way to cut down a lot of dependency bloat I think? E.g. I don't use Svelte, Vue, or Sass, together already 20 MB.

@natemoo-re maybe something similar can be done as [your idea of bundling dependencies in microsite](https://github.com/natemoo-re/microsite/issues/95) 😉

**Comment by natemoo-re** [comment]
I'd be in favor of this! We've been meaning to remove the renderers by default before v1.0, and this move would make sense with the work @FredKSchott is doing on integrations. Would love to pair this with a focused effort on improving our dependency graph.

**Reply by Siilwyn** [reply]
Thanks for the reply & great to hear this in on the radar!

**Comment by aFuzzyBear** [comment]
TYPESCRIPT 😱😱😱😱

**Reply by natemoo-re** [reply]
Yeah this is included because `astro check` needs it, but I'm not sure if we really want `astro check` to be part of core since it could easily be an add-on package `astro-check`.

In fact, `astro check` could just prompt you to install `astro-check` as a separate devDependency if it's missing.

**Comment by jonathantneal** [comment]
I like this effort. I think it will be good to also [compare packages with their cores](https://packagephobia.com/result?p=astro%2Cnext%2Cnuxt%2Cvite).

| package | version | development size | publish size |
|:--------|--------:|-----------------:|-------------:|
| astro   | 0.22.20 |            190MB |       12.9MB |
| next    | 12.0.10 |            137MB |       29.9MB |

| package | version | development size | publish size |
|:--------|--------:|-----------------:|-------------:|
| vite    |  2.7.13 |           24.1MB |       11.6MB |
| webpack |  5.68.0 |           14.9MB |       4.13MB |

**Comment by natemoo-re** [comment]
@Siilwyn would you be interested in leading this effort in collaboration with us? You've identified a few big targets that would be great to eliminate—I expect this to be a medium-to-large effort, so we'd love all the help we can get!

**Reply by natemoo-re** [reply]
To elaborate more on my comment about integrations... I expect removing the renderers from core will be a smoother transition once we have the API in place for folks to do something `astro add --renderer preact`. 

Until then, there's definitely some low-hanging fruit we can tackle. I'd love to dig into _why_ some of the largest dependencies are here and see if we can address them while reducing our footprint.

**Reply by Siilwyn** [reply]
Hey sounds good! I'd be up for leading this. It will be a slow burner, I'll mostly be active on fridays but I'd be happy to help :)

I'll start with looking into the biggest dependencies, should I open the outcome in an RFC, issue or PR?
